### ISSUE TITLE
v5.5.0+ Update breaks build on MetricsController.init()

### ISSUE THREAD

**Issue by richzeng** [mixpanel/mixpanel-android#594]
I tried building my app locally for the first time in about a month and noticed that it was crashing on bootup when making the call to `MetricsController.init()`. I suspect this is due to the removal of GCM in Mixpanel.

Here's the stacktrace.

```
 java.lang.NoSuchMethodError: No virtual method getInstanceId()Lcom/google/android/gms/tasks/Task; in class Lcom/google/firebase/iid/FirebaseInstanceId; or its super classes (declaration of 'com.google.firebase.iid.FirebaseInstanceId' appears in /data/app/com.wearhaus.frontend-Ujj1M5P0smL5sHuYSLLgQA==/split_lib_dependencies_apk.apk!classes2.dex)
        at com.mixpanel.android.mpmetrics.MixpanelFCMMessagingService.init(MixpanelFCMMessagingService.java:117)
        at com.mixpanel.android.mpmetrics.MixpanelAPI.getInstance(MixpanelAPI.java:423)
        at com.mixpanel.android.mpmetrics.MixpanelAPI.getInstance(MixpanelAPI.java:369)
        at com.wearhaus.frontend.controllers.MetricsController.init(MetricsController.java:42)
        at com.wearhaus.frontend.controllers.WearhausApp.setupFacebook(WearhausApp.java:132)
        at com.wearhaus.frontend.controllers.WearhausApp.onAppLaunched(WearhausApp.java:122)
        at com.wearhaus.frontend.controllers.WearhausApp.enforceAppLaunched(WearhausApp.java:101)
        at com.wearhaus.frontend.account.LoginActivity.onCreate(LoginActivity.java:76)
        at android.app.Activity.performCreate(Activity.java:7144)
        at android.app.Activity.performCreate(Activity.java:7135)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2931)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

Specifying Mixpanel v5.4.5 in my `build.gradle` solved the issue for me. I have a few questions and concerns.
1. Will doing staying at v5.4.5 have any negative consequences for my Mixpanel experience? We've had it installed and working well for 2 years without having to touch it. We'd rather not spend the time updating all our Gradle dependencies at this time.
2. What's the logic behind the documentation's instructions to put `mixpanel-android:5.+` instead of a specific version number in `build.gradle`? It goes against Android best practices and issues like what I encountered are exactly why.
3. Was there any kind of announcement about this change? I didn't see anything about it and it seems like the kind of breaking API update that should be announced.

**Comment by patedit** [comment]
Hi @richzeng - sorry for the inconvenience. I'll address your issues one by one:

1. We encourage customers to use the latest version of the SDK since it includes new functionality and bug fixes. From v5.4.5 to v5.6.0 we have added some cool stuff like FCM support, event-triggered in-app notification support and account level analytics! Take a look at the releases notes [here](https://github.com/mixpanel/mixpanel-android/releases).
2. We will get that changed soon.
3. For those customers affected, the SDK prints warning logs describing how to migrate from GCM to FCM. It shouldn't crash the app. We did write [release notes](https://github.com/mixpanel/mixpanel-android/releases/tag/v5.5.0) and they also appear on our [changelog](https://github.com/mixpanel/mixpanel-android/wiki/Changelog). I am curious, where did you try to find this announcement? We can start using other channels, let me know what you think! 

For your particular problem, it does seem that you have included on your `AndroidManifest.xml` references to firebase, but you haven't included the library on your `build.gradle`. Is that the case?
We do not initialize FCM unless you have `com.google.firebase.MESSAGING_EVENT` intent.
